File: address-geocode-canada-dot-net-examples/GetPostalCodeInfoRestSdkExample.cs
using address_geocode_canada_dot_net.REST;

namespace address_geocode_ca_dot_net_examples
{
    public class GetPostalCodeInfoRestSdkExample
    {
        public static void Go(string LicenseKey, bool IsLive)
        {
            Console.WriteLine("\r\n-------------------------------------------------------");
            Console.WriteLine("Address Geocode – Canada - GetPostalCodeInfo - REST SDK");
            Console.WriteLine("-------------------------------------------------------");

            GetPostalCodeInfoClient.GetPostalCodeInfoInput getPostalCodeInfoInput = new(
                PostalCode: "K1Y 4E9",
                LicenseKey: LicenseKey,
                IsLive: IsLive,
                TimeoutSeconds: 15
            );

            Console.WriteLine("\r\n* Input *\r\n");
            Console.WriteLine($"Postal Code: {getPostalCodeInfoInput.PostalCode}");
            Console.WriteLine($"License Key: {getPostalCodeInfoInput.LicenseKey}");
            Console.WriteLine($"Is Live    : {getPostalCodeInfoInput.IsLive}");

            GetPostalCodeInfoResponse response = GetPostalCodeInfoClient.Invoke(getPostalCodeInfoInput);
            if (response.Error is null)
            {
                Console.WriteLine("\r\n* Geocode Info *\r\n");
                Console.WriteLine($"Latitude      : {response.Latitude}");
                Console.WriteLine($"Longitude     : {response.Longitude}");
                Console.WriteLine($"PostalCode    : {response.PostalCode}");
                Console.WriteLine($"TimeZone      : {response.TimeZone}");
                Console.WriteLine($"DST           : {response.DST}");
                Console.WriteLine($"AreaCode      : {response.AreaCode}");
                Console.WriteLine($"City          : {response.City}");
                Console.WriteLine($"CityPopulation: {response.CityPopulation}");
                Console.WriteLine($"Province      : {response.Province}");
            }
            else
            {
                Console.WriteLine("\r\n* Error *\r\n");
                Console.WriteLine($"Error Desc    : {response.Error.Desc}");
                Console.WriteLine($"Error Number  : {response.Error.Number}");
                Console.WriteLine($"Error Location: {response.Error.Location}");
            }
        }
    }
}

File: address-geocode-canada-nodejs-examples/get_postal_code_info_rest_sdk_example.js
import { GetPostalCodeInfoClient } from "../address-geocode-canada-nodejs/REST/get_postal_code_info_rest.js";

async function GetPostalCodeInfoRestGo(licenseKey, isLive) {
    console.log("\n-------------------------------------------------------");
    console.log("Address Geocode - Canada - GetPostalCodeInfo - REST SDK");
    console.log("-------------------------------------------------------");

    const postalCode = "N2E 1P4";
    const timeoutSeconds = 15;

    console.log("\n* Input *\n");
    console.log(`PostalCode     : ${postalCode}`);
    console.log(`License Key    : ${licenseKey}`);
    console.log(`Is Live        : ${isLive}`);
    console.log(`Timeout Seconds: ${timeoutSeconds}`);

    try {
        const response = await GetPostalCodeInfoClient.invoke(
            postalCode,
            licenseKey,
            isLive,
            timeoutSeconds
        );

        console.log("\n* Geocode Info *\n");
        if (response) {
            console.log(`Latitude      : ${response.Latitude}`);
            console.log(`Longitude     : ${response.Longitude}`);
            console.log(`PostalCode    : ${response.PostalCode}`);
            console.log(`TimeZone      : ${response.TimeZone}`);
            console.log(`DST           : ${response.DST}`);
            console.log(`AreaCode      : ${response.AreaCode}`);
            console.log(`City          : ${response.City}`);
            console.log(`CityPopulation: ${response.CityPopulation}`);
            console.log(`Province      : ${response.Province}`);
        } else {
            console.log("No geocode info found.");
        }

        if (response.Error) {
            console.log("\n* Error *\n");
            console.log(`Error Desc    : ${response.Error.Desc}`);
            console.log(`Error Number  : ${response.Error.Number}`);
            console.log(`Error Location: ${response.Error.Location}`);
            return;
        }
    } catch (e) {
        console.log("\n* Error *\n");
        console.log(`Error Message: ${e.message}`);
    }
}

export { GetPostalCodeInfoRestGo };
